nengo_loihi/synapses.py

```diff
@@ -356,3 +363,113 @@ def validate(self):
         if self.pop_type == 16:
             if self.axon_cx_bases is not None:
                 assert np.all(self.axon_cx_bases % 4 == 0)
+
+
+class SynapseGroup(object):
+    def __init__(self):
+        self.synapses = []
+        self.named_synapses = {}
+
+    # def __contains__(self, obj):
+    #     return obj in (self.synapses if isinstance(obj, Synapses) else
+    #                    self.named_synapses)
+
+    # def __getitem__(self, key):
+    #     return self.named_synapses[key]
```

Comment:
Leftover comments

Reply:
I left this in to remind me to talk about it. I was originally going to use `__contains__` and `__getitem__` to look for and get values from `named_synapses`. I ended up making the more explicit `by_name` and `has_name`, instead. Does this make sense? Using `__contains__` and `__getitem__` is definitely slicker, but then they don't match `__iter__` (which iterates over all synapses, not just those in `named_synapses`).

Reply:
Yeah that reasoning makes sense to me, seems like that could definitely lead to some sneaky bugs if `__contains__` and `__iter__` are referring to different collections.